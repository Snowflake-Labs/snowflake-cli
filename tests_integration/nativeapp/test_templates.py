import pytest


@pytest.mark.integration
def test_list_templates_no_options_success(runner, snapshot):
    args = ["app", "list-templates"]
    result = runner.invoke_json(args)

    assert result.exit_code == 0
    templates = result.json
    assert len(templates) > 0
    assert "basic" in [t["template"] for t in templates]
