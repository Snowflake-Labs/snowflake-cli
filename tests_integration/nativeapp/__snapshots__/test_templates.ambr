# serializer version: 1
# name: test_list_templates_no_options_success
  '''
  +------------------------------------------------------------------------------+
  | template         | description                                               |
  |------------------+-----------------------------------------------------------|
  | basic            | This is the basic project template for a Snowflake Native |
  |                  | App project. It contains minimal code meant to help you   |
  |                  | set up your first application object in your account      |
  |                  | quickly.                                                  |
  | streamlit-java   | This is an example template for a Snowflake Native App    |
  |                  | project which demonstrates the use of Java extension code |
  |                  | and adding Streamlit code. This template is meant to      |
  |                  | guide developers towards a possible project structure on  |
  |                  | the basis of functionality, as well as to indicate the    |
  |                  | contents of some common and useful files.                 |
  | streamlit-python | This is an example template for a Snowflake Native App    |
  |                  | project which demonstrates the use of Python extension    |
  |                  | code and adding Streamlit code. This template is meant to |
  |                  | guide developers towards a possible project structure on  |
  |                  | the basis of functionality, as well as to indicate the    |
  |                  | contents of some common and useful files.                 |
  +------------------------------------------------------------------------------+
  
  '''
# ---
